contracts/Removal.sol

```diff
@@ -65,40 +53,117 @@ contract Removal is
     _setApprovalForAll(owner, operator, approved);
   }
 
+  function createRemovalId(bytes calldata removalData)
```

Comment:
This block of functions feels very redundant but I agree with the end result that they become part of `Removal`'s public API.  The rest of them after this one might also feel tidied up if they used the `removalId.version()` style naming in their implementations.

Reply:
definitely feels a bit cleaned up this way.... I also removed "FromRemovalId" from the name of all these functions, which is what I think your comment above was suggesting, so that we're just writing `removalId.fieldWeWant()` and it's less redundant